Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,125 @@
+        }
+    }
+
+    var errorDescription: String? {
+        description
+    }
+
+    /// The error is definitely due to misconfiguration of a target.
+    var isDefiniteMisconfigurationError: Bool {
+        switch self {
+        case .incompatibleTarget:
+            return false
+        case .noTargetOrDependenciesWithExpectedFiles:
+            return false
+        case .badArguments:
+            return false
+        case .noTargetsMatchingTargetName:
+            return false
+        case .tooManyTargetsMatchingTargetName:
+            return false
+        case .fileErrors(let errors):
+            return errors.isDefiniteMisconfigurationError
+        }
+    }
+}
+
+struct FileError: Swift.Error, CustomStringConvertible, LocalizedError {
+
+    enum Kind: CaseIterable {
+        case config
+        case document
+    }
+
+    enum Issue {
+        case noFilesFound
+        case multipleFilesFound(files: [Path])
+
+        /// The error is definitely due to misconfiguration of a target.
+        var isDefiniteMisconfigurationError: Bool {
+            switch self {
+            case .noFilesFound:
+                return false
+            case .multipleFilesFound:
+                return true
+            }
+        }
+    }
+
+    let targetName: String
+    let fileKind: Kind
+    let issue: Issue
+
+    var description: String {
+        "FileError { \(helpAnchor!) }"
+    }
+
+    var helpAnchor: String? {
+        switch fileKind {
+        case .config:
+            switch issue {
+            case .noFilesFound:
+                return "No config file found in the target named '\(targetName)'. Add a file called 'openapi-generator-config.yaml' or 'openapi-generator-config.yml' to the target's source directory. See documentation for details."
+            case .multipleFilesFound(let files):
+                return "Multiple config files found in the target named '\(targetName)', but exactly one is expected. Found \(files.map(\.description).joined(separator: " "))."
+            }
+        case .document:
+            switch issue {
+            case .noFilesFound:
+                return "No OpenAPI document found in the target named '\(targetName)'. Add a file called 'openapi.yaml', 'openapi.yml' or 'openapi.json' (can also be a symlink) to the target's source directory. See documentation for details."
+            case .multipleFilesFound(let files):
+                return "Multiple OpenAPI documents found in the target named '\(targetName)', but exactly one is expected. Found \(files.map(\.description).joined(separator: " "))."
+            }
+        }
+    }
+
+    var errorDescription: String? {
+        description
+    }
+
+    static func notFoundFileErrors(forTarget targetName: String) -> [FileError] {
+        FileError.Kind.allCases.map { kind in
+            FileError(targetName: targetName, fileKind: kind, issue: .noFilesFound)
+        }
+    }
+}
+
+private extension [FileError] {
```

Comment:
```suggestion
private extension Array where Element == FileError {
```

Nit: This is the spelling we use in this project.